### ISSUE TITLE
Error upon installing

### ISSUE THREAD

**Issue by Pokeylooted** [Icy-Thought/snowflake#28]
Hey when doing the install I keep getting this error *this is on fresh minimal install*:
       ```
error: The option `programs.regreet.settings.GTK.cursor_theme_name' has conflicting definition values:
       - In `/nix/store/q2fj92xzn1m0f5n3q9jhmhxk6cq92bcz-source': "Bibata-Modern-Classic"
       - In `/nix/store/d9gbq853jvbccrz5g3y0irbwgc57w137-source/nixos/modules/programs/regreet.nix': "Adwaita"
       Use `lib.mkForce value` or `lib.mkDefault value` to change the priority on any of these definitions.
```

**Comment by Icy-Thought** [comment]
Well, I haven't really used regreet after trying it out since its intentions was for wayland use. (which I have been sleeping on because of XMonad)

You can either replace the value for line 237 in `snowflake/modules/themes/default.nix` with:
```nix
          cursor_theme_name = lib.mkForce "${pointer.name}";
```

or just remove that line altogether.

**Comment by Pokeylooted** [comment]
> Well, I haven't really used regreet after trying it out since its intentions was for wayland use. (which I have been sleeping on because of XMonad)
> 
> 
> 
> You can either replace the value for line 237 in `snowflake/modules/themes/default.nix` with:
> 
> ```nix
> 
>           cursor_theme_name = lib.mkForce "${pointer.name}";
> 
> ```
> 
> 
> 
> or just remove that line altogether.

Hey thanks for the reply! I was wondering where I can put things for systemd? Currently it tries to install grub

**Comment by Icy-Thought** [comment]
For the bootloader just remove the grub bits and replace with systemd-boot if that is what you are looking for.
https://nixos.wiki/wiki/Bootloader

**Comment by Icy-Thought** [comment]
What script?

**Comment by Pokeylooted** [comment]
> What script?

The flake* I noticed it was trying to install grub, ill see if I can catch it again

**Comment by Pokeylooted** [comment]
Oh! Also another question is this a bug or defining this in configuration.nix and rebuilding seems to not get rid of this message unless i do the environmnet variable?
```
       error: Package ‘spicetify-catppuccin’ in /nix/store/d9gbq853jvbccrz5g3y0irbwgc57w137-source/pkgs/applications/audio/spotify/default.nix:14 has an unfree license (‘unfree’), refusing to evaluate.

       a) To temporarily allow unfree packages, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNFREE=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnfree = true; }
       in configuration.nix to override this.

       Alternatively you can configure a predicate to allow specific packages:
         { nixpkgs.config.allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) [
             "spotify"
           ];
         }

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnfree = true; }
       to ~/.config/nixpkgs/config.nix.
```

**Comment by Icy-Thought** [comment]
Well here is what you are doing wrong for starters, you are not reading the error messages. It mentions that you have disabled non-free packages. (1st line)

**Comment by Icy-Thought** [comment]
Also you don't have grub disabled, did you check what the default nixos option for that is [Link](https://search.nixos.org/options?channel=24.05&show=boot.loader.grub.enable&from=0&size=50&sort=relevance&type=packages&query=grub)

**Comment by Pokeylooted** [comment]
> Well here is what you are doing wrong for starters, you are not reading the error messages. It mentions that you have disabled non-free packages. (1st line)

I have it enabled, I looked in my /etc/nixos/configuration.nix and rebuild

**Comment by Pokeylooted** [comment]
> Also you don't have grub disabled, did you check what the default nixos option for that is [Link](https://search.nixos.org/options?channel=24.05&show=boot.loader.grub.enable&from=0&size=50&sort=relevance&type=packages&query=grub)

You enabled grub in default.nix that's what I was mainly asking where grub was enabled at

**Comment by Icy-Thought** [comment]
Yes i did, hence I mentioned what lines you should be looking at if you are interested in disabling it altogether.

**Comment by Icy-Thought** [comment]
Are you building from your `/etc/nixos/configuration.nix` or my repo through the appropriate command for your system?

**Comment by Pokeylooted** [comment]
> Are you building from your `/etc/nixos/configuration.nix` or my repo through the appropriate command for your system?

Your repo, I was wondering what config changes I need to for xmonad to work? I enabled it however when I get to lightdm it doesn't let me login as it's looking for gnome

**Comment by Pokeylooted** [comment]
I looked at xmonad.nix in desktop/skeleton, and it seems likes its not using lightdm there? However, because of the default x11 set to lightdm its being used there which is breaking it?

**Comment by Icy-Thought** [comment]
What skeleton bit are you trying to enable? If it's x11 based then you need to rewrite a bit of `themes/default.nix` and `modules/desktop/default.nix`.

**Comment by Icy-Thought** [comment]
I can always test it out since I was interested in using regreet at first, but I don't know what kept me from fully migrating tbh.

**Comment by Pokeylooted** [comment]
> What skeleton bit are you trying to enable? If it's x11 based then you need to rewrite a bit of `themes/default.nix` and `modules/desktop/default.nix`.

By default gnome is enabled which is Wayland, and enables regreet. However I wanted to try out qmonad, but when I enable it and goto the login screen it tries to look for gnome as the session

**Comment by Icy-Thought** [comment]
That is because I haven't written a case for multiple DE/WM's since i have never done that tbh. Gotta refactor stuff if you want to go down that route.

**Comment by Pokeylooted** [comment]
> That is because I haven't written a case for multiple DE/WM's since i have never done that tbh. Gotta refactor stuff if you want to go down that route.

Are you currently using snowflake? If so could I look at the part of code that you have for xmonad? Since you mentioned that you're using it. Also I'm not trying to use multiple DE/WMs just trying to try out xmonad

**Comment by Icy-Thought** [comment]
Everything I am using is atm pushed to this repo. But I haven't enabled multiple DE/WM 's and therefore I don't know how it would currently work with my setup.

**Comment by Pokeylooted** [comment]
> Everything I am using is atm pushed to this repo. But I haven't enabled multiple DE/WM 's before and therefore I don't know how it would currently work with my setup.

Wdym multiple de? How did you enable qmonad for your setup?

**Comment by Pokeylooted** [comment]
All I have done is replace the line: `gnome.enable = true` to `xmonad.enable = true;`

**Comment by Icy-Thought** [comment]
I assumed you were since you mentioned Gnome and XMonad and having them enabled, I guess I misinterpreted your previous messages. What about my current config is not working for you? And how are you installing your system?

**Comment by Pokeylooted** [comment]
> I assumed you were since you mentioned Gnome and XMonad and having them enabled, I guess I misinterpreted your previous messages. What about my current config is not working for you? And how are you installing your system?

The current config tries to use lightdm, and lightdm doesn't find xmonad so I can't login I get stuck in that screen, I was just wondering how you got past that. I'm installing my system by doing a minimal install then installing flake

**Comment by Icy-Thought** [comment]
Why is lightdm not finding XMonad?  Aren't you trying to replicate my setup? I am confused.

**Comment by Pokeylooted** [comment]
> Why is lightdm not finding XMonad?  Aren't you trying to replicate my setup? I am confused.

Not sure why it's not finding xmonad, and yes I was replicating your setup. Upon the logging screen after I enter password it just says can't find desktop session
